src/libcore/num/mod.rs

```diff
@@ -464,6 +452,30 @@ macro_rules! uint_impl {
                     v => Some(self / v),
                 }
             }
+
+            #[inline]
+            fn pow(self, mut exp: u32) -> Self {
+                let mut base = self;
+                let mut acc: Self = Int::one();
+                let mut did_base_overflow = false;
+
+                while exp > 0 {
+                    if (exp & 1) == 1 {
+                        acc = acc * base;
+                        if did_base_overflow {
+                            panic!("overflow detected");
+                        }
```

Comment:
That's I what I couldn't completely understand : if we panic, why should we
give attention to the returned result ?
Le 24 févr. 2015 09:54, "Felix S Klock II" <notifications@github.com> a
écrit :

> In src/libcore/num/mod.rs
> <https://github.com/rust-lang/rust/pull/22087#discussion_r25236519>:
>
> > @@ -464,6 +452,30 @@ macro_rules! uint_impl {
> >                      v => Some(self / v),
> >                  }
> >              }
> > +
> > +            #[inline]
> > +            fn pow(self, mut exp: u32) -> Self {
> > +                let mut base = self;
> > +                let mut acc: Self = Int::one();
> > +                let mut did_base_overflow = false;
> > +
> > +                while exp > 0 {
> > +                    if (exp & 1) == 1 {
> > +                        acc = acc * base;
> > +                        if did_base_overflow {
> > +                            panic!("overflow detected");
> > +                        }
>
> (Having said that, I prefer my encoding of this semantics, given in the
> gist linked above)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/22087/files#r25236519>.
>